lib/ask-stack-result-view.coffee

```diff
@@ -179,13 +185,24 @@ class AskStackResultView extends ScrollView
         codeHl =  hljs.highlightAuto(code.text()).value
         code.html(codeHl)
 
-  addCodeButtons: (elem_id) ->
+  addCodeButtons: (elem_id, id, id_type) ->
+    console.log(id, id_type);
     pres = @resultsView.find("##{elem_id}").find('pre')
     for pre in pres
-      btnInsert = @genCodeButton('Insert')
+      btnInsert = @genCodeButton('Insert', id, id_type)
       $(pre).prev().after(btnInsert)
 
-  genCodeButton: (type) ->
+  genCodeButton: (type, id, id_type) ->
+    console.log(id, id_type);
+    # Attribute author
+    if id != undefined
+      author_src = $("##{id_type}-author-link-#{id}").attr('href');
+      author_name = $("##{id_type}-author-link-#{id}").html();
+      source_src = $("##{id_type}-link-#{id}").attr('href');
+      qa = true;
```

Comment:
It was basically a check to make sure that an id was passed in, just in case anywhere else was also using that variable. `qa` meant that a question/answer id was passed in and we have generated the author/source links. I've now moved this since it made more sense to do the check based off just the id instead of making a flag. I'll mark the line once I've pushed so that you can see what I mean.